indexing-hadoop/src/main/java/io/druid/indexer/path/PathSpec.java

```diff
@@ -28,13 +28,14 @@
 
 /**
  */
-@JsonTypeInfo(use= JsonTypeInfo.Id.NAME, property="type")
-@JsonSubTypes(value={
-    @JsonSubTypes.Type(name="granular_unprocessed", value=GranularUnprocessedPathSpec.class),
-    @JsonSubTypes.Type(name="granularity", value=GranularityPathSpec.class),
-    @JsonSubTypes.Type(name="static", value=StaticPathSpec.class),
-    @JsonSubTypes.Type(name="dataSource", value=DatasourcePathSpec.class),
-    @JsonSubTypes.Type(name="multi", value=MultiplePathSpec.class)
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name= "granular_unprocessed", value = GranularUnprocessedPathSpec.class),
+    @JsonSubTypes.Type(name= "granularity", value = GranularityPathSpec.class),
+    @JsonSubTypes.Type(name= "static", value = StaticPathSpec.class),
+    @JsonSubTypes.Type(name= "dataSource", value = DatasourcePathSpec.class),
+    @JsonSubTypes.Type(name= "multi", value = MultiplePathSpec.class),
+    @JsonSubTypes.Type(name= "partition", value = PartitionPathSpec.class)
```

Comment:
Well, another point that this patch provides is to get dimension values, not stored in files actually, from directory paths.
That's the reason why it is named PartitionPathSpec and only applicable to the directory structure for partitioned table of SQL-on-Hadoop solutions.
I think keeping both PartitionPathSpec and StaticPathSpec is better.